code/modules/mob/living/carbon/human/inventory.dm

```diff
@@ -62,12 +62,27 @@ This saves us from having to call add_fingerprint() any time something is put in
 	var/obj/item/organ/external/O = organs_by_name[name]
 	return (O && !O.is_stump() && (!check_usablility || O.is_usable()))
 
-/mob/living/carbon/human/u_equip(obj/item/W as obj)
+/mob/living/carbon/human/u_equip(obj/item/Item)
+	if(src.client)
+		src.client.screen -= Item
+	Item.layer = initial(Item.layer)
+	Item.plane = initial(Item.plane)
+	Item.screen_loc = null
+
+	legacy_u_equip(Item)
```

Comment:
This is silly, why would we have a function that calls legacy_(itself)

Reply:
But that's legacy `u_equip` it will be removed lately, but that can't be done now.

Reply:
sure it can, just put that code back where it came from

Infact i've already done it, its in master now, so you can just completely erase your changes to that function and leave u_equip as is

Reply:
But there is only half of changes in your PR, not icon handling, for example. 
I think there is no real reason to mess item variable changing with all that slots related code.
Code will be cleaner if we keep both procs for now. It's even not break compability with anything.